src/main/java/com/example/demo/service/impl/BookServiceImpl.java

```
@@ -33,4 +34,16 @@ public BookDto getById(Long id) {
         return bookMapper.toDto(bookRepository.findById(id).orElseThrow(
                 () -> new EntityNotFoundException("Can not get book by id:" + id)));
     }
+
+    @Override
+    public void update(CreateBookRequestDto createBookRequestDto, Long id) {
+        Book book = bookMapper.toModel(createBookRequestDto);
+        book.setId(id);
```

check if **id** exists and throw an exception in case it is invalid